pennylane/tape/qscript.py

```diff
@@ -546,25 +551,51 @@ def trainable_params(self, param_indices):
 
         self._trainable_params = sorted(set(param_indices))
 
-    def get_operation(self, idx):
+    def get_operation(self, idx, return_op_index=False):
         """Returns the trainable operation, and the corresponding operation argument
         index, for a specified trainable parameter index.
 
         Args:
             idx (int): the trainable parameter index
-
+            return_op_index (bool): The function also returns the operation index.
         Returns:
             tuple[.Operation, int]: tuple containing the corresponding
             operation, and an integer representing the argument index,
             for the provided trainable parameter.
```

Comment:
We should update this as well.  Now it could also return `tuple[.Operator, int, int]` as well, depending on the `return_op_index` keyword argument.